mlflow/dspy/autolog.py

```diff
@@ -66,6 +85,37 @@ def autolog(
     def trace_disabled_fn(original, self, *args, **kwargs):
         # NB: Since calling mlflow.dspy.autolog() again does not unpatch a function, we need to
         # check this flag at runtime to determine if we should generate traces.
+        @trace_disabled
+        def _fn(self, *args, **kwargs):
+            return original(self, *args, **kwargs)
+
+        def _compile_fn(self, *args, **kwargs):
+            if callback := _active_callback():
+                callback._within_compile = True
```

Comment:
nit: this should be public attr since you are assigning values to it.

Reply:
Yeah, I was in two minds since we do both ways in mlflow repo, let me change it to a public attr here.